self_test/faceted_surface_vtu_output/tetmesh_faceted_surfaces.h

```diff
@@ -0,0 +1,1256 @@
+   box_point[2] = offset[2]-box_half_length;
+   Vertex_pt[5]=new TetMeshVertex(box_point);
+   
+   box_point[0] = offset[0]+box_half_width;
+   box_point[1] = offset[1]+box_half_width;
+   box_point[2] = offset[2]+box_half_length;
+   Vertex_pt[6]=new TetMeshVertex(box_point);
+   
+   box_point[0] = offset[0]+box_half_width;
+   box_point[1] = offset[1]-box_half_width;
+   box_point[2] = offset[2]+box_half_length;
+   Vertex_pt[7]=new TetMeshVertex(box_point);
+   
+
+   // Make facets
+   unsigned n_facet=6;
+   Facet_pt.resize(n_facet);
+   
+   unsigned n_vertex_on_facet=4;
+   Facet_pt[0]=new TetMeshFacet(n_vertex_on_facet);
+   unsigned one_based_boundary_id=1+one_based_boundary_id_offset;
+   Facet_pt[0]->set_one_based_boundary_id(one_based_boundary_id);   
+   Facet_pt[0]->set_vertex_pt(0,Vertex_pt[0]);
+   Facet_pt[0]->set_vertex_pt(1,Vertex_pt[4]);
+   Facet_pt[0]->set_vertex_pt(2,Vertex_pt[7]);
+   Facet_pt[0]->set_vertex_pt(3,Vertex_pt[3]);
+
+   Facet_pt[1]=new TetMeshFacet(n_vertex_on_facet);
+   one_based_boundary_id=2+one_based_boundary_id_offset;
+   Facet_pt[1]->set_one_based_boundary_id(one_based_boundary_id); 
+   Facet_pt[1]->set_vertex_pt(0,Vertex_pt[4]);
+   Facet_pt[1]->set_vertex_pt(1,Vertex_pt[5]);
+   Facet_pt[1]->set_vertex_pt(2,Vertex_pt[6]);
+   Facet_pt[1]->set_vertex_pt(3,Vertex_pt[7]);
+
+   // top
+   Facet_pt[2]=new TetMeshFacet(n_vertex_on_facet);
+   one_based_boundary_id=3+one_based_boundary_id_offset;
+   Facet_pt[2]->set_one_based_boundary_id(one_based_boundary_id); 
+   Facet_pt[2]->set_vertex_pt(0,Vertex_pt[3]);
+   Facet_pt[2]->set_vertex_pt(1,Vertex_pt[7]);
+   Facet_pt[2]->set_vertex_pt(2,Vertex_pt[6]);
+   Facet_pt[2]->set_vertex_pt(3,Vertex_pt[2]);
+
+   Facet_pt[3]=new TetMeshFacet(n_vertex_on_facet);
+   one_based_boundary_id=4+one_based_boundary_id_offset;
+   Facet_pt[3]->set_one_based_boundary_id(one_based_boundary_id);   
+   Facet_pt[3]->set_vertex_pt(0,Vertex_pt[6]);
+   Facet_pt[3]->set_vertex_pt(1,Vertex_pt[5]);
+   Facet_pt[3]->set_vertex_pt(2,Vertex_pt[1]);
+   Facet_pt[3]->set_vertex_pt(3,Vertex_pt[2]);
+
+   Facet_pt[4]=new TetMeshFacet(n_vertex_on_facet);
+   one_based_boundary_id=5+one_based_boundary_id_offset;
+   Facet_pt[4]->set_one_based_boundary_id(one_based_boundary_id);  
+   Facet_pt[4]->set_vertex_pt(0,Vertex_pt[3]);
+   Facet_pt[4]->set_vertex_pt(1,Vertex_pt[2]);
+   Facet_pt[4]->set_vertex_pt(2,Vertex_pt[1]);
+   Facet_pt[4]->set_vertex_pt(3,Vertex_pt[0]);
+
+   // bottom
+   Facet_pt[5]=new TetMeshFacet(n_vertex_on_facet);
+   one_based_boundary_id=6+one_based_boundary_id_offset;
+   Facet_pt[5]->set_one_based_boundary_id(one_based_boundary_id);   
+   Facet_pt[5]->set_vertex_pt(0,Vertex_pt[5]);
+   Facet_pt[5]->set_vertex_pt(1,Vertex_pt[1]);
+   Facet_pt[5]->set_vertex_pt(2,Vertex_pt[0]);
+   Facet_pt[5]->set_vertex_pt(3,Vertex_pt[4]);
+
+
+   // Specify the adjacent region that is bounded by these facets
+   for (unsigned f=0;f<6;f++)
+    {
```

Comment:
Hmm, more formatting oddities; the braces don't seem to align with the `f` on the preceding line and no spaces around operators...